symphony/app/fbcnms-projects/workflows/src/routes.js

```diff
@@ -20,553 +20,570 @@ import moment from 'moment';
+  );
+
+  router.get(
+    '/metadata/workflow/:name/:version',
+    async (req: ExpressRequest, res, next) => {
+      try {
+        const result = await http.get(
+          baseURLMeta +
+            'workflow/' +
+            req.params.name +
+            '?version=' +
+            req.params.version,
+          req,
+        );
+        res.status(200).send({result});
+      } catch (err) {
+        next(err);
+      }
+    },
+  );
 
-const findSchedule = (schedules, name) => {
-  for (const schedule of schedules) {
-    if (schedule.name === name) {
-      return schedule;
+  router.put('/metadata', async (req: ExpressRequest, res, next) => {
+    try {
+      const result = await http.put(baseURLMeta + 'workflow/', req.body, req);
+      res.status(200).send(result);
+    } catch (err) {
+      res.status(400).send(err.response.body);
+      next(err);
     }
-  }
-  return null;
-};
+  });
+
+  // Conductor only allows POST for event handler creation
+  // and PUT for updating. This code works around the issue by
+  // trying PUT if POST fails.
+  router.post('/event', async (req: ExpressRequest, res, next) => {
+    try {
+      const result = await http.post(eventURL, req.body, req);
+      res.status(200).send(result);
+    } catch (err) {
+      logger.info(`Got exception ${JSON.stringify(err)} while POSTing event`);
+      try {
+        const result = await http.put(eventURL, req.body, req);
+        res.status(200).send(result);
+      } catch (e) {
+        logger.info(`Got exception ${JSON.stringify(e)} while PUTting event`);
+        res.status(400).send('Post and Put failed');
+        next(e);
+      }
+    }
+  });
 
-router.get('/metadata/workflow', async (req: ExpressRequest, res, next) => {
-  try {
-    const result = await http.get(baseURLMeta + 'workflow', req);
-    // combine with schedules
-    const schedules = await http.get(baseURLSchedule, req);
-    for (const workflowDef of result) {
-      const expectedScheduleName = workflowDef.name + ':' + workflowDef.version;
-      const found = findSchedule(schedules, expectedScheduleName);
-      workflowDef.hasSchedule = found != null;
-      workflowDef.expectedScheduleName = expectedScheduleName;
+  router.post('/workflow', async (req: ExpressRequest, res, next) => {
+    try {
+      const result = await http.post(baseURLWorkflow, req.body, req);
+      res.status(200).send(result);
+    } catch (err) {
+      next(err);
     }
+  });
 
-    res.status(200).send({result});
-  } catch (err) {
-    next(err);
-  }
-});
+  router.get('/executions', async (req: ExpressRequest, res, next) => {
+    try {
+      const freeText = [];
+      if (req.query.freeText !== '') {
+        freeText.push(req.query.freeText);
+      } else {
+        freeText.push('*');
+      }
 
-router.delete(
-  '/metadata/workflow/:name/:version',
-  async (req: ExpressRequest, res, next) => {
+      let h: string = '-1';
+      if (req.query.h !== 'undefined' && req.query.h !== '') {
```

Comment:
did you mean `typeof req.query.h`, or did you mean to check that `h` isn't the string literal `"undefined"`?

Reply:
I haven't actually changed this code so it's probably correct. I just moved all the code under a function. This github diff probably doesn't align - and + lines correctly.

Reply:
I haven't actually changed this code so it's probably correct. I just moved all the code under a function. This github diff probably doesn't align - and + lines correctly.